### DISCUSSION TITLE
[Chapter 04] Spring MVC -> Spring Boot

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#18] [Chapter 04. Spring Web MVC Overview] [answered]
책에 내용을 흐름에 따라 읽던 중
스프링 부트의 장점에 대해 간략하게 소개가 나옵니다.

1. 간단하게 설정하고 바로 서비스 투입이 가능하다.
2. 내장 서버를 지원한다 (Tomcat, Jetty, Undertow)
3. spring-boot-starter

제가 spring-boot-starter에 대해 모르기전에는 mybatis 또는 jdbc를 일반적으로 Spring MVC에서 하듯 그냥 일반 라이브러리 의존성을 가져와서 환경은 부트인데 제가 하나하나 Connection부터 만들어 삽질을 한 경험들이 있습니다.
이런 것처럼 기존 spring mvc보다 진화한 boot에서만 충분히 지원하고있는 장점들에 대해 더 알려주시면 감사하겠습니다.

**Comment by Irisation23** [comment, marked as answer]
대표적으로 `CP`에 대해 빈 등록을 해주지 않아도 됩니다.

스프링 부트는 `HikariCP`를 기본 `CP`로 사용하고 있습니다.
그리고 `CP`의 개수를 따로 커스텀 해주지 않는다면 defualt 개수는 10개 입니다.

**Comment by coalong** [comment]
저도 이 책에서 소개하는 spring boot 의 장점을 읽으면서 단순히 설정의 간소화를 위해 나온게 아니라,
MSA 를 위해 Spring boot 의 등장은 필수적이라 생각했습니다.

 제가 spring boot 기반 프로젝트를 했을 당시 느꼈던 장점은
 spring cloud 를 사용하여 클라우드 환경을 손쉽게 구성하고 여러 마이크로 서비스를 구축, 운영하기에 수월했습니다.
